src/app/server/Server.cpp

```diff
@@ -136,7 +136,8 @@ CHIP_ERROR RestoreAllAdminPairingsFromKVS(AdminPairingTable & adminPairings, Adm
         }
         else
         {
-            ChipLogProgress(AppServer, "Found admin pairing for %d, node ID %llu", admin->GetAdminId(), admin->GetNodeId());
+            ChipLogProgress(AppServer, "Found admin pairing for %d, node ID %08lx%08lx", admin->GetAdminId(),
+                            static_cast<unsigned long>(admin->GetNodeId() >> 32), static_cast<unsigned long>(admin->GetNodeId()));
```

Comment:
is there a possibility that unsigned long is 64-bit, in which case the 2nd %08lx will print the entire node id? I believe LP64 systems (linux?) will behave like this.

How about having some macro/helper that also includes a `& 0xFFFFFFFF` in the cast?

Reply:
Or just cast to uint32_t and use the right printf format for that?

Reply:
agreed - changed to uint32_t casts and using PRIx32 specifiers